Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java

```diff
@@ -175,13 +200,52 @@ private CharSequence handleAnchor(
             PostLinkable pl = new PostLinkable(theme, handlerLink.key, handlerLink.value, handlerLink.type);
             res.setSpan(pl, 0, res.length(), (250 << Spanned.SPAN_PRIORITY_SHIFT) & Spanned.SPAN_PRIORITY);
             post.addLinkable(pl);
+            spannableStringBuilder.append(res);
 
-            return res;
+            return spannableStringBuilder;
         } else {
             return null;
         }
     }
 
+    private void addMockReply(
+            Theme theme,
+            Post.Builder post,
+            SpannableStringBuilder spannableStringBuilder,
+            int postNo,
+            MockReplyManager.MockReply mockReply
+    ) {
+        Logger.d(TAG, "Adding a new mock reply (replyTo: " + mockReply.getPostNo() + ", replyFrom: " + postNo + ")");
+        post.addReplyTo(mockReply.getPostNo());
+
+        CharSequence replyText = String.format(
+                Locale.US,
+                ">>%d%s",
+                mockReply.getPostNo(),
+                MOCK_REPLY_SUFFIX
+        );
+
+        SpannableString res = new SpannableString(replyText);
+
+        PostLinkable pl = new PostLinkable(
+                theme,
+                replyText,
+                mockReply.getPostNo(),
+                PostLinkable.Type.QUOTE
+        );
+
+        res.setSpan(
+                pl,
+                0,
+                res.length(),
+                (250 << Spanned.SPAN_PRIORITY_SHIFT) & Spanned.SPAN_PRIORITY
+        );
+
+        post.addLinkable(pl);
+        spannableStringBuilder.append(res);
```

Comment:
You can chain appends. That's kinda the point of a builder.

Reply:
Done.